projects/debouncer/src/debouncer.c

```diff
@@ -0,0 +1,54 @@
+#include "debouncer.h"
+
+#include <stddef.h>
+#include "soft_timer.h"
+
+#define DEBOUNCER_INTERRUPT_MASKING_DURATION_MS 50
+
+// This is the callback for the soft timer. If there is a button input, it runs the user's callback.
+static void prv_timer_callback(SoftTimerID timer_id, void *context) {
+  DebouncerInfo *debouncer_info = context;
+
+  GPIOState current_state;
+  gpio_get_state(&debouncer_info->address, &current_state);
+  if (!debouncer_info->callback) return;
```

Comment:
Two things:
1) We don't use single-line conditionals like this
2) This won't have the intended behavior - by returning early, you'll miss reenabling the interrupt.